### DISCUSSION TITLE
why is modern pixijs an order of magnitude slower than pixijs v3.0.0?

### DISCUSSION THREAD

**Discussion by farzher** [pixijs/pixijs#8181] [Q&A]
this is bunnymark. it renders 100k sprites at 17fps https://pixijs.io/bunny-mark/

this is an old version of bunnymark using pixi v3. it renders 100k sprites at 165 fps (at least) https://www.goodboydigital.com/pixijs/bunnymark/

i'm trying to render a lot of sprites, but i can't figure out how to go fast.

**Comment by bigtimebuddy** [comment]
Reposted from Slack where someone asked something similar recently:

----

This is a complicated answer.

The original bunnymark on http://goodboydigital.com is using a very old version of PixiJS (v3) and that bunny-mark is designed for a narrowly optimized demo. In other words, for pushing as many bunnies to the stage as possible on a single base texture. This isn't exactly a "real-world" experience. Over time, bunny-mark has evolved as more of a way to measure realistic performance relatively between releases than it is showing off. If you want the fasted single experience, then use a ParticleContainer which should get you something closer to the v3 version.

These tools were designed with different purposes. That being said, the newer version is expected to be slightly slower for a few reasons that are intentional (transpiling, more robust batching, optimized for real-world conditions).

**Comment by farzher** [comment]
ok `ParticleContainer` helps! it's at 88 fps for me now. still slower but only 2x as slow instead of 10x

would you happen to know if there's anyway to simply render a texture onto the screen? i remember asking this in the discord a while back and i think the answer is no. but all i need to do is simply render a texture to the screen, many times. i don't need any of the features of the Sprite class, i do need optimized rendering

**Reply by bigtimebuddy** [reply]
It all depends on what your scene is, how complicated a thing you’re trying to build, how often it changes, and what other things are running/rendering concurrently. 

The more you get to the edges of performance, the more the specifics make a huge difference how you approach optimization. There is not magic bullet solution for rendering. 

Are you doing any culling? Visible elements offscreen still need transform updates, which can be costly at 100K.

Are you sure you need 100K? Fill-rate can be a performance killer. So if you have elements that’d are overlapping other elements, you pay a cost for rendering those hidden pixels.

**Reply by ivanpopelyshev** [reply]
Make a Mesh, there's nothing that can be faster than it.
